libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Trace.hs

```diff
@@ -370,28 +404,36 @@ instance
          in trace
               (raiseMockError lastSlot nextSlotNo epochstate pdfs txsl gs)
               (error "FAILS")
-      -- Left pdfs -> trace ("Discard\n"++show(ppList (ppMockChainFailure reify) pdfs)) discard -- TODO should we enable this?
       Right mcs2 -> seq mcs2 (pure mockblock)
 
-  shrinkSignal (MockBlock _ _ xs) | SS.null xs = []
-  shrinkSignal (MockBlock i s xs) = [MockBlock i s (SS.drop 1 xs), MockBlock i s (SS.take (SS.length xs - 1) xs)]
+  shrinkSignal _ = []
 
-mapProportion :: (v -> Int) -> Map.Map k v -> Gen k
-mapProportion toInt m =
+--  shrinkSignal (MockBlock _ _ xs) | SS.null xs = []
+--  shrinkSignal (MockBlock i s xs) = [MockBlock i s (SS.drop 1 xs), MockBlock i s (SS.take (SS.length xs - 1) xs)]
+
+mapProportion :: (EpochNo, Word64, Int) -> (v -> Int) -> Map.Map k v -> Gen k
+mapProportion (epochnum, lastSlot, count) toInt m =
```

Comment:
Why is it a tripple? Also why pass function instead of `fmap`ping it directly over a map prior to passing it.
```suggestion
mapProportion :: EpochNo -> Word64 -> Int -> Map.Map k Int -> Gen k
mapProportion epochNum lastSlot count m =
```

Reply:
done